rubygem-smart_proxy_dynflow_core/rubygem-smart_proxy_dynflow_core.spec

```diff
@@ -0,0 +1,148 @@
+Name: %{?scl_prefix}rubygem-%{gem_name}
+Version: 0.1.2
+Release: 1%{?foremandist}%{?dist}
+Group: Development/Libraries
+License: GPLv3
+URL: http://github.com/theforeman/smart_proxy_dynflow
+Source0: http://rubygems.org/downloads/%{gem_name}-%{version}.gem
+Requires: foreman-proxy >= 1.11.0
+
+Requires: %{?scl_prefix}rubygem(bundler_ext)
+Requires: %{?scl_prefix}rubygem(dynflow) >= 0.8.4
+Requires: %{?scl_prefix}rubygem(dynflow) < 0.9.0
+Requires: %{?scl_prefix}rubygem(sequel)
+Requires: %{?scl_prefix}rubygem(rest-client)
+Requires: %{?scl_prefix_ror}rubygem(sinatra) >= 1.4
+Requires: %{?scl_prefix_ror}rubygem(sinatra) < 2.0
+Requires: %{?scl_prefix_ror}rubygem(rack)
+Requires: %{?scl_prefix_ror}rubygem(sqlite3)
+Requires: %{?scl_prefix_ruby}ruby(release)
+Requires: %{?scl_prefix_ruby}rubygems
+%if 0%{?rhel} == 6
+Requires(post): chkconfig
+Requires(preun): chkconfig
+Requires(preun): initscripts
+%else
+Requires(post): systemd-sysv
+Requires(post): systemd-units
+Requires(preun): systemd-units
+BuildRequires: systemd
+%endif
+
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+BuildRequires: %{?scl_prefix_ruby}rubygems
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+
+BuildArch: noarch
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+
+%description
+Use the Dynflow inside Foreman smart proxy
+
+%package doc
+BuildArch:  noarch
+Requires:   %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+Summary:    Documentation for rubygem-%{gem_name}
+
+%description doc
+This package contains documentation for rubygem-%{gem_name}.
+
+%prep
+%setup -n %{pkg_name}-%{version} -q -c -T
+mkdir -p .%{gem_dir}
+%{?scl:scl enable %{scl} - <<EOF}
+%gem_install -n %{SOURCE0}
+%{?scl:EOF}
+
+%build
+sed -ri '1sX(/usr/bin/ruby|/usr/bin/env ruby)X/usr/bin/%{?scl:%{scl_prefix}}rubyX' ./%{gem_instdir}/bin/%{service_name}
+
+# switches to bundler_ext instead of bundler
+mv %{_builddir}/%{pkg_name}-%{version}/%{gem_instdir}/Gemfile ./%{gem_instdir}/Gemfile.in
+sed -ri 'sX\#\{File\.dirname\(__FILE__\)\}/bundler\.d/\*\.rbX%{_datadir}/%{gem_name}/bundler.d/*.rbX' ./%{gem_instdir}/Gemfile.in
+
+# switch to SCL directory for config
+sed -ri 'sX/etcX%{_sysconfdir}X' ./%{gem_instdir}/lib/%{gem_name}/*.rb
+
+%install
+mkdir -p %{buildroot}%{_datadir}/%{gem_name}/bundler.d
+mkdir -p %{buildroot}%{gem_dir}
+cp -a .%{gem_dir}/* %{buildroot}%{gem_dir}/
+mkdir -p %{buildroot}%{_bindir}
+cp -pa .%{_bindir}/smart_proxy_dynflow_core %{buildroot}%{_bindir}/smart_proxy_dynflow_core
+mkdir -p %{buildroot}%{_sysconfdir}/smart_proxy_dynflow_core
+cp -pa .%{gem_instdir}/config/settings.yml.example %{buildroot}%{_sysconfdir}/smart_proxy_dynflow_core/settings.yml
+
+#copy init scripts and sysconfigs
+%if 0%{?rhel} == 6
+install -Dp -m0755 %{buildroot}%{gem_instdir}/deploy/%{service_name}.init %{buildroot}%{_root_initddir}/%{service_name}
+%else
+install -Dp -m0644 %{buildroot}%{gem_instdir}/deploy/%{service_name}.service %{buildroot}%{_unitdir}/%{service_name}.service
```

Comment:
This unit file (and also the init script) contains SCL specifics which won't work on Fedora and shouldn't really be in the source.  I'd prefer:

1. Remove the scl enable from the scripts/unit files
2. Move the `smart_proxy_dynflow_core` binstub from the SCL bindir to the root bindir, like we do with Hammer, so it's installed at /usr/bin/smart_proxy_dynflow_core.  (Like the above discussion about configs, this means it's not contained in the SCL.)
3. Fix the comment at https://github.com/theforeman/foreman-packaging/pull/1150/files#r67669806 so the binstub gets the /usr/bin/tfm-ruby shebang line, then the unit file doesn't need to enable the SCL.

or

1. Put separate SCL/non-SCL unit files into either the gem or the packaging repo (preferably, since everything's OS-specific).